docs/cli_usage_guide.md

```diff
@@ -0,0 +1,149 @@
+# Using the firewatch CLI
+
+## Table of Contents
+
+* [Using the firewatch CLI](#using-the-firewatch-cli)
+  * [Installation](#installation)
+    * [Docker (recommended)](#docker-recommended)
+    * [Local Machine (using venv)](#local-machine-using-venv)
+  * [Configuration](#configuration)
+    * [Jira issue creation (`firewatch report`) configuration](#jira-issue-creation-firewatch-report-configuration)
+      * [Defining the Configuration](#defining-the-configuration)
+  * [Usage](#usage)
+    * [`report`](#report)
+    * [`jira_config_gen`](#jiraconfiggen)
+
+## Installation
+
+### Docker (recommended)
+
+1. Ensure you have [Docker installed](https://www.docker.com/get-started/) on your system.
+2. Clone the repository: `git clone https://github.com/CSPI-QE/firewatch.git`.
+3. Navigate to the project root in your terminal: `cd firewatch`.
+4. Run the following to build and run a Docker container with firewatch installed: `make build-run`.
+5. Use the `firewatch` command to execute the tool. See the [CLI usage guide](docs/cli_usage_guide.md) for instructions on using the tool.
+
+### Local Machine (using venv)
+
+1. Clone the repository: `git clone https://github.com/CSPI-QE/firewatch.git`
+2. Navigate to the project root: `cd firewatch`
+3. Install the necessary dependencies: `make dev-environment`
+4. Use the `firewatch` command to execute the tool. See the [CLI usage guide](docs/cli_usage_guide.md) for instructions on using the tool.
+
+## Configuration
+
+### Jira issue creation (`firewatch report`) configuration
+
+Firewatch was designed to allow for users to define which Jira issues get created depending on which failures are found in a OpenShift CI failed run. Using an easy-to-define JSON config, users can easily track issues in their OpenShift CI runs efficiently.
+
+**Example:**
+
+```json
+[
+  {"step": "exact-step-name", "failure_type": "pod_failure", "classification": "Infrastructure", "jira_project": "PROJECT"},
+  {"step": "*partial-name*", "failure_type": "all", "classification":  "Misc.", "jira_project": "OTHER"},
+  {"step": "*ends-with-this", "failure_type": "test_failure", "classification": "Test failures", "jira_project": "TEST"},
+  {"step": "*ignore*", "failure_type": "test_failure", "classification": "NONE", "jira_project": "NONE", ignore: "true"}
+]
+```
+
```

Comment:
I am wondering if we can add the additional JIRA issue parameters like priority, labels etc, to the JSON ?

Reply:
It is certainly possible, I think that would probably be a good extension of this. For now, I think the current work serves the purpose we need. Maybe after I get the scenarios on-boarded to it, I will make that change.

Unless you think, we need to add it before any of the scenarios start using the tool. If so, I can start working on an update.

Reply:
@arajesh when you have been manually creating tickets based on failures have you been using labels, components or anything like that to help you track these things?

@calebevans I also believe adding these things can be considered and RFE and done in a separate PR later on.

Reply:
Yes those changes can be added later. I just wanted to know if those are supported. Thanks!